core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -307,8 +311,8 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
 
   // Determine block size constrained by shared memory:
   inline unsigned local_block_size(const FunctorType& f) {
-    // FIXME_HIP I don't know where 8 comes from
-    unsigned int n = ::Kokkos::Experimental::Impl::HIPTraits::WarpSize * 8;
+    unsigned int n =
+        ::Kokkos::Experimental::Impl::HIPTraits::MaxThreadsPerBlock;
```

Comment:
This was previously 64*8=512. Did you change it on purpose to 1024?

Reply:
It's more for consistency (and removing magic constants) than anything else.  1024 should be valid and perform fine (as long as you're not in a case where you would spill w/ 1024, but not w/ 512... but that's what the LaunchBounds are for :))

Reply:
Moved this bit over to !3299